### PR TITLE
Fix bug during response evaluation

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- fixes https://github.com/RasaHQ/rasa/issues/5916

If the test data is missing a response that is present in the training data and during testing this response is predicted, the evaluation failed. This is not fixed.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by dakshvar22** [comment]
@tabergma Tried reproducing the functionality, attaching a screenshot - 
response_selection_confusion_matrix

I am thinking if we should refactor our train test split function for response selector data to always include all responses both in train and test splits. That way, if a user uses `rasa data split`, they'll never run into this problem. What do you think?

**Comment by tabergma** [comment]
Was thinking the same, but I guess we should create a separate issue for that. Would nevertheless keep this "fix" as it also might happen, that users split the data on their own and miss some labels in the test set. What do you think?

**Comment by dakshvar22** [comment]
Makes sense, do we need a changelog entry for it?

**Comment by tabergma** [comment]
I don't think a changelog entry is needed as the feature is not yet released. Will create a separate issue for the train/test split.
